demonstrations/tutorial_qcbm.py

```diff
@@ -0,0 +1,327 @@
+    # such that the leaves of the TTN are at the measurement,
+    # more faithfully mimicking a TTN.
+    qml.adjoint(
+        qml.TTN(
+            wires=range(QUBITS),
+            n_block_wires=2,
+            block=block,
+            n_params_block=6,
+            template_weights=template_weights,
+        )
+    )
+
+    return qml.probs()
+
+
+##############################################################################
+# You will notice that we do not need to embed any data into the circuit, since QCBMs are generative models!
+# Any interaction with our dataset will take place in the cost function.
+# Now, we define our loss, which, as mentioned before, will be the KL-Divergence loss:
+
+
+def kl_div(p, q):
+    return np.sum(q * np.log(q / p))
+
+
+##############################################################################
+# We are now ready to train some circuits! Let's generate our dataset first and prepare the optimizer.
+idxs, true_probs = prepare_dataset(DATASET_SIZE, 4, 4)
+optimizer = qml.AdamOptimizer()
+
+
+##############################################################################
+# Let's start with the MPS circuit:
+def cost_fn_mps(weights, idxs, true_probs):
+    probs = qcbm_circuit_mps(weights)
+    pred_probs = probs[idxs]
+    return kl_div(pred_probs, true_probs)
+
+
+mps_weights_init = np.random.random(size=MPS_DATA_SHAPE)
+mps_weights = mps_weights_init
+
+mps_costs = []
+for it in range(TRAINING_ITERATIONS + 1):
+    mps_weights = optimizer.step(cost_fn_mps, mps_weights, idxs=idxs, true_probs=true_probs)
+
+    current_cost = cost_fn_mps(mps_weights, idxs, true_probs)
+
+    mps_costs.append(current_cost)
+
+    if it % 10 == 0:
+        print(f"Iter: {it:4d} | KL Loss: {current_cost:0.7f}")
+
+##############################################################################
+# Now we do the same for the TTN network, resetting the optimizer first.
+optimizer.reset()
+
+
+def cost_fn_ttn(weights, idxs, true_probs):
+    probs = qcbm_circuit_ttn(weights)
+    pred_probs = probs[idxs]
+    return kl_div(pred_probs, true_probs)
+
+
+ttn_weights_init = np.random.random(size=TTN_DATA_SHAPE)
+ttn_weights = ttn_weights_init
+
+ttn_costs = []
+for it in range(TRAINING_ITERATIONS + 1):
+    ttn_weights = optimizer.step(cost_fn_ttn, ttn_weights, idxs=idxs, true_probs=true_probs)
+
+    current_cost = cost_fn_ttn(ttn_weights, idxs, true_probs)
+
+    ttn_costs.append(current_cost)
+
+    if it % 10 == 0:
+        print(f"Iter: {it:4d} | KL Loss: {current_cost:0.7f}")
+
+##############################################################################
+# Let's now plot the loss curves of the experiments against each other
+
+plt.plot(range(TRAINING_ITERATIONS + 1), mps_costs, ".b", label="MPS Losses")
+plt.plot(range(TRAINING_ITERATIONS + 1), ttn_costs, ".g", label="TTN Losses")
+plt.xlabel("Iteration #", fontsize=16)
+plt.ylabel("KL Loss", fontsize=16)
+plt.legend()
+plt.show()
+
```

Comment:
If I got it right, after you train the QCBMs models you should be able to generate some prob. distribution right? Why not showing that too! That's a way to show that your models works well and I think it will make the demo more beautiful  and self-content :)